sklearn/linear_model/_ridge.py

```diff
@@ -1842,13 +1844,42 @@ def fit(self, X, y, sample_weight=None):
         if sample_weight is not None:
             sample_weight = _check_sample_weight(sample_weight, X, dtype=X.dtype)
 
-        self.alphas = np.asarray(self.alphas)
-
-        if np.any(self.alphas <= 0):
-            raise ValueError(
-                "alphas must be strictly positive. Got {} containing some "
-                "negative or null value instead.".format(self.alphas)
+        if isinstance(self.alphas, (np.ndarray, list, tuple)):
+            n_alphas = 1 if np.ndim(self.alphas) == 0 else len(self.alphas)
```

Comment:
As discussed with @ArturoAmorQ , this provides correct parameter validation to all sub-classes of `_RidgeGCV`.

Yet, in order for the validation to be run once, we need, for each subclass:
 - to move those validation at the beginning of `fit`
 - to move the core of `_fit` in a dedicated private method (`_fit`) which does all but the validation
 - to move other validation steps as well (?)
 - to check that the validation is only done once.
 
For us, this small refactoring has be done in another PR.

cc @glemaitre: we need your authority on the best way to proceed.

Reply:
Yes, this is indeed what we are leaning towards as a solution. We already applied such a pattern with @jeremiedbb: validating in `fit` that will call `_fit` where the processing is done with no validation.